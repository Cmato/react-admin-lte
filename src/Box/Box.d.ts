import * as React from 'react';

export interface BoxProps {
  collapsed?: boolean;
  loading?: boolean;
  loadingIndicator?: React.ReactNode;
  solid?: boolean;
  style?: BoxStyle;
}

export type BoxStyle = 'default' | 'primary' | 'success' | 'warning' | 'danger';

declare const Box: React.ComponentType<BoxProps>;

export default Box;
