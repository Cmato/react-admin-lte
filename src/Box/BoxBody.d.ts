import * as React from 'react';

export interface BoxBodyProps {
  noPadding?: boolean;
}

declare const BoxBody: React.ComponentType<BoxBodyProps>;

export default BoxBody;
