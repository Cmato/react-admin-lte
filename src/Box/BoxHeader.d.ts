import * as React from 'react';

export interface BoxHeaderProps {
  border?: boolean;
}

declare const BoxHeader: React.ComponentType<BoxHeaderProps>;

export default BoxHeader;
