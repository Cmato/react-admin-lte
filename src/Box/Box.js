import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Box = ({
  children,
  className: classNameProp,
  collapsed,
  loading,
  loadingIndicator,
  solid,
  style,
  ...other
}) => {
  const className = classNames({
    'box': true,
    'collapsed-box': collapsed,
    [`box-${style}`]: style,
    'box-solid': solid
  }, classNameProp);

  const overlay = loading
    ? <div className="overlay">{loadingIndicator}</div>
    : null;

  return (
    <div className={className} {...other}>
      {children}
      {overlay}
    </div>
  );
};

Box.propTypes = {
  /**
   * The content of the box.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * If `true`, collapses the box, otherwise expands it.
   */
  collapsed: PropTypes.bool,
  /**
   * If `true', `loadingIndicator` is overlaid on top of box.
   */
  loading: PropTypes.bool,
  /**
   * To be shown when `loading` is `true`.
   */
  loadingIndicator: PropTypes.node,
  /**
   * Displays box with solid colored header.
   */
  solid: PropTypes.bool,
  /**
   * Contextual style of the box.
   */
  style: PropTypes.oneOf(['default', 'primary', 'success', 'warning', 'danger'])
};

Box.defaultProps = {
  collapsed: false,
  loading: false,
  solid: false
};

export default Box;
