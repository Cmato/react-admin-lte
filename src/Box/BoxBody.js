import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const BoxBody = ({
  children,
  className: classNameProp,
  noPadding,
  ...other
}) => {
  const className = classNames({
    'box-body': true,
    'no-padding': noPadding
  }, classNameProp);

  return (
    <div className={className} {...other}>
      {children}
    </div>
  );
};

BoxBody.propTypes = {
  /**
   * The contents of the box body.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string,
  /**
   * If `true`, padding will be removed from the box body.
   */
  noPadding: PropTypes.bool
};

BoxBody.defaultProps = {
  noPadding: false
};

export default BoxBody;
