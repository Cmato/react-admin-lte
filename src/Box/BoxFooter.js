import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const BoxFooter = ({
  children,
  className: classNameProp,
  ...other
}) => {
  const className = classNames({
    'box-footer': true
  }, classNameProp);

  return (
    <div className={className} {...other}>
      {children}
    </div>
  );
};

BoxFooter.propTypes = {
  /**
   * The contents of the box footer.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string
};

export default BoxFooter;
