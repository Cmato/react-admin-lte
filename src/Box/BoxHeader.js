import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const BoxHeader = ({
  border,
  children,
  className: classNameProp,
  ...other
}) => {
  const className = classNames({
    'box-header': true,
    'with-border': border
  }, classNameProp);

  return (
    <div className={className} {...other}>
      {children}
    </div>
  );
};

BoxHeader.propTypes = {
  /**
   * If `true`, header will have a bottom border.
   */
  border: PropTypes.bool,
  /**
   * The contents of the box header.
   */
  children: PropTypes.node.isRequired,
  /**
   * @ignore
   */
  className: PropTypes.string
};

BoxHeader.defaultProps = {
  border: false
};

export default BoxHeader;
