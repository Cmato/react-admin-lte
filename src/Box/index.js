export { default } from './Box';
export { default as BoxBody } from './BoxBody';
export { default as BoxFooter } from './BoxFooter';
export { default as BoxHeader } from './BoxHeader';
export { default as BoxTitle } from './BoxTitle';
export { default as BoxTools } from './BoxTools';
