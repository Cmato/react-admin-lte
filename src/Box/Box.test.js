import React from 'react';
import { shallow } from 'enzyme';
import Box from './Box';

it('renders default', () => {
  expect(shallow(<Box><div /></Box>))
    .toMatchSnapshot();
});

it('renders with style', () => {
  expect(shallow(<Box style="default"><div /></Box>))
    .toMatchSnapshot();
});

it('renders solid', () => {
  expect(shallow(<Box solid><div /></Box>))
    .toMatchSnapshot();
});

it('renders loading', () => {
  expect(shallow(<Box loading><div /></Box>))
    .toMatchSnapshot();
});

it('renders loading indicator', () => {
  expect(shallow(<Box loading loadingIndicator={<div>spin</div>}><div /></Box>))
    .toMatchSnapshot();
});

it('renders collapsed', () => {
  expect(shallow(<Box collapsed><div /></Box>))
    .toMatchSnapshot();
});
