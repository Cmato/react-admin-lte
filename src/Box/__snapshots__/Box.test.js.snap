// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders collapsed 1`] = `
<div
  className="box collapsed-box"
>
  <div />
</div>
`;

exports[`renders default 1`] = `
<div
  className="box"
>
  <div />
</div>
`;

exports[`renders loading 1`] = `
<div
  className="box"
>
  <div />
  <div
    className="overlay"
  />
</div>
`;

exports[`renders loading indicator 1`] = `
<div
  className="box"
>
  <div />
  <div
    className="overlay"
  >
    <div>
      spin
    </div>
  </div>
</div>
`;

exports[`renders solid 1`] = `
<div
  className="box box-solid"
>
  <div />
</div>
`;

exports[`renders with style 1`] = `
<div
  className="box box-default"
>
  <div />
</div>
`;
